src/assets/scripts/client/airport/MapCollection.js

```diff
@@ -0,0 +1,180 @@
+         * @default #_items.length
+         */
+
+        this._init(mapJson, defaultMaps, airportPositionModel, magneticNorth);
+    }
+
+    /**
+     * A flag indicating whether the `MapCollection` has any visible maps.
+     *
+     * @for MapCollection
+     * @property hasVisibleMaps
+     * @return {boolean}
+     */
+    get hasVisibleMaps() {
+        return this._items.some((map) => !map.isHidden);
+    }
+
+    /**
+     * Public fascade for `#_items`
+     *
+     * @for MapCollection
+     * @property maps
+     * @return {array<MapModel>}
+     */
+    get maps() {
+        return this._items;
+    }
+
+    /**
+     * A list of all the map lines for the visible maps in the `MapCollection`
+     *
+     * @for MapCollection
+     * @method getVisibleMapLines
+     * @return {array<object>}
+     */
+    getVisibleMapLines() {
+        return this._items.reduce((sum, map) => {
+            if (map.isHidden) {
+                return sum;
+            }
+
+            return [
+                ...map.lines,
+                ...sum
+            ];
+        }, []);
+    }
+
+    /**
+     * A list of names of all the visible maps in the `MapCollection`
+     *
+     * @for MapCollection
+     * @method getVisibleMapNames
+     * @return {array<object>}
+     */
+    getVisibleMapNames() {
+        return this._items.reduce((sum, map) => {
+            if (map.isHidden) {
+                return sum;
+            }
+
+            return [
+                ...sum,
+                map.name
+            ];
+        }, []);
+    }
+
+    /**
+     * A list of all the `MapModel` names in the `MapCollection`
+     *
+     * @for MapCollection
+     * @method getMapNames
+     * @returns {array<string>}
+     */
+    getMapNames() {
+        return this._items.map((map) => map.name);
+    }
+
+    /**
+     * Sets which maps should be rendered on the `CanvasController`
+     *
+     * @for MapCollection
+     * @method setVisibleMaps
+     * @param names {array<string>}
+     */
+    setVisibleMaps(names) {
+        this._items.forEach((map) => {
+            map.isHidden = !names.includes(map.name);
+        });
+    }
+
+    /**
+     * Initialize the instance
+     *
+     * @for MapCollection
+     * @method _init
+     * @param mapJson {object}
+     * @param defaultMaps {array<string>}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    _init(mapJson, defaultMaps, airportPositionModel, magneticNorth) {
+        this._buildMapModels(mapJson, defaultMaps, airportPositionModel, magneticNorth);
+    }
+
+    /**
+     * @for MapCollection
+     * @method _buildMapModels
+     * @param mapJson {object}
+     * @param defaultMaps {array<string>}
+     * @param airportPositionModel {StaticPositionModel}
+     * @param magneticNorth {number}
+     */
+    _buildMapModels(mapJson, defaultMaps, airportPositionModel, magneticNorth) {
+        this._items = mapJson.map((item) => {
```

Comment:
`return mapJson.map(.....` instead of setting this._items

Because generally where we see "build", "generate", or "calculate" in method names, they are only assembling or calculating the answer and returning it, rather than actually using the answer.